<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>hopEditor</class>
 <widget class="QDialog" name="hopEditor">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>778</width>
    <height>450</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Hop Editor</string>
  </property>
  <property name="configSection" stdset="0">
   <string notr="true">hopEditor</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QTabWidget" name="tabWidget_editor">
     <property name="tabPosition">
      <enum>QTabWidget::West</enum>
     </property>
     <property name="tabShape">
      <enum>QTabWidget::Rounded</enum>
     </property>
     <property name="currentIndex">
      <number>0</number>
     </property>
     <property name="usesScrollButtons">
      <bool>false</bool>
     </property>
     <property name="movable">
      <bool>true</bool>
     </property>
     <widget class="QWidget" name="tabWidget_editorPage1">
      <attribute name="title">
       <string>Main</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout">
       <item row="0" column="2">
        <widget class="QLabel" name="label_use">
         <property name="toolTip">
          <string>Required</string>
         </property>
         <property name="text">
          <string>Use</string>
         </property>
         <property name="buddy">
          <cstring>comboBox_use</cstring>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QLabel" name="label_name">
         <property name="toolTip">
          <string>Required</string>
         </property>
         <property name="text">
          <string>Name</string>
         </property>
         <property name="buddy">
          <cstring>lineEdit_name</cstring>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label_alpha">
         <property name="toolTip">
          <string>Required</string>
         </property>
         <property name="text">
          <string>Alpha %</string>
         </property>
         <property name="buddy">
          <cstring>lineEdit_alpha</cstring>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="BtGenericEdit" name="lineEdit_alpha">
         <property name="toolTip">
          <string>Alpha acids as percent by mass</string>
         </property>
         <property name="editField" stdset="0">
          <string notr="true">alpha_pct</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="BtStringEdit" name="lineEdit_name">
         <property name="toolTip">
          <string>Name</string>
         </property>
        </widget>
       </item>
       <item row="7" column="0">
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="0" column="3">
        <widget class="QComboBox" name="comboBox_use">
         <property name="toolTip">
          <string>Use</string>
         </property>
         <item>
          <property name="text">
           <string>Mash</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>First Wort</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Boil</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Aroma</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Dry Hop</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="1" column="2">
        <widget class="QLabel" name="label_type">
         <property name="text">
          <string>Type</string>
         </property>
         <property name="buddy">
          <cstring>comboBox_type</cstring>
         </property>
        </widget>
       </item>
       <item row="1" column="3">
        <widget class="QComboBox" name="comboBox_type">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="toolTip">
          <string>Type</string>
         </property>
         <item>
          <property name="text">
           <string>Bittering</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Aroma</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Both</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="2" column="2">
        <widget class="QLabel" name="label_form">
         <property name="text">
          <string>Form</string>
         </property>
         <property name="buddy">
          <cstring>comboBox_form</cstring>
         </property>
        </widget>
       </item>
       <item row="2" column="3">
        <widget class="QComboBox" name="comboBox_form">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="toolTip">
          <string>Form</string>
         </property>
         <item>
          <property name="text">
           <string>Leaf</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Pellet</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Plug</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="BtMassEdit" name="lineEdit_inventory">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="toolTip">
          <string>Amount in inventory</string>
         </property>
         <property name="editField" stdset="0">
          <string notr="true">inventory</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="BtMassLabel" name="label_inventory">
         <property name="contextMenuPolicy">
          <enum>Qt::CustomContextMenu</enum>
         </property>
         <property name="text">
          <string>Amount in Inventory</string>
         </property>
         <property name="buddy">
          <cstring>lineEdit_inventory</cstring>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tabWidget_editorPage2">
      <attribute name="title">
       <string>Extras</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="0" column="2">
        <widget class="BtTimeLabel" name="label_time">
         <property name="contextMenuPolicy">
          <enum>Qt::CustomContextMenu</enum>
         </property>
         <property name="text">
          <string>Time</string>
         </property>
         <property name="buddy">
          <cstring>lineEdit_time</cstring>
         </property>
        </widget>
       </item>
       <item row="6" column="1">
        <widget class="BtGenericEdit" name="lineEdit_cohumulone">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="toolTip">
          <string>Cohumulone</string>
         </property>
         <property name="editField" stdset="0">
          <string notr="true">cohumulone_pct</string>
         </property>
        </widget>
       </item>
       <item row="7" column="1">
        <widget class="BtGenericEdit" name="lineEdit_myrcene">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="toolTip">
          <string>Myrcene</string>
         </property>
         <property name="editField" stdset="0">
          <string notr="true">myrcene_pct</string>
         </property>
        </widget>
       </item>
       <item row="8" column="0">
        <spacer name="verticalSpacer_3">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="6" column="0">
        <widget class="QLabel" name="label_cohumulone">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>Cohumulone %</string>
         </property>
         <property name="buddy">
          <cstring>lineEdit_cohumulone</cstring>
         </property>
        </widget>
       </item>
       <item row="7" column="0">
        <widget class="QLabel" name="label_myrcene">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>Myrcene %</string>
         </property>
         <property name="buddy">
          <cstring>lineEdit_myrcene</cstring>
         </property>
        </widget>
       </item>
       <item row="0" column="3">
        <widget class="BtTimeEdit" name="lineEdit_time">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="toolTip">
          <string>Time</string>
         </property>
         <property name="editField" stdset="0">
          <string notr="true">time_min</string>
         </property>
        </widget>
       </item>
       <item row="1" column="2">
        <widget class="QLabel" name="label_HSI">
         <property name="text">
          <string>HSI</string>
         </property>
         <property name="buddy">
          <cstring>lineEdit_HSI</cstring>
         </property>
        </widget>
       </item>
       <item row="1" column="3">
        <widget class="BtGenericEdit" name="lineEdit_HSI">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="toolTip">
          <string>Hop Stability/Storage index</string>
         </property>
         <property name="editField" stdset="0">
          <string notr="true">hsi_pct</string>
         </property>
        </widget>
       </item>
       <item row="2" column="2">
        <widget class="QLabel" name="label_origin">
         <property name="text">
          <string>Origin</string>
         </property>
         <property name="buddy">
          <cstring>lineEdit_origin</cstring>
         </property>
        </widget>
       </item>
       <item row="2" column="3">
        <widget class="BtStringEdit" name="lineEdit_origin">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="toolTip">
          <string>Origin</string>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QLabel" name="label_beta">
         <property name="text">
          <string>Beta %</string>
         </property>
         <property name="buddy">
          <cstring>lineEdit_beta</cstring>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="BtGenericEdit" name="lineEdit_beta">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="toolTip">
          <string>Beta acids as percent by mass</string>
         </property>
         <property name="editField" stdset="0">
          <string notr="true">beta_pct</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label_humulene">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>Humulene %</string>
         </property>
         <property name="buddy">
          <cstring>lineEdit_humulene</cstring>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="BtGenericEdit" name="lineEdit_humulene">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="toolTip">
          <string>Humulene</string>
         </property>
         <property name="editField" stdset="0">
          <string notr="true">humulene_pct</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="label_caryophyllene">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>Caryophyllene %</string>
         </property>
         <property name="buddy">
          <cstring>lineEdit_caryophyllene</cstring>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="BtGenericEdit" name="lineEdit_caryophyllene">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="toolTip">
          <string>Caryophyllene %</string>
         </property>
         <property name="editField" stdset="0">
          <string notr="true">caryophyllene_pct</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tabWidget_editorPage4">
      <attribute name="title">
       <string>Substitutes</string>
      </attribute>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <widget class="QTextEdit" name="textEdit_substitutes">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="MinimumExpanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>0</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>1000</width>
           <height>1000</height>
          </size>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tabWidget_editorPage5">
      <attribute name="title">
       <string>Notes</string>
      </attribute>
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <item>
        <widget class="QTextEdit" name="textEdit_notes">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="MinimumExpanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>100</width>
           <height>100</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>1000</width>
           <height>1000</height>
          </size>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_3">
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="pushButton_new">
       <property name="toolTip">
        <string>New hop</string>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="icon">
        <iconset resource="../brewken.qrc">
         <normaloff>:/images/smallPlus.svg</normaloff>:/images/smallPlus.svg</iconset>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="pushButton_save">
       <property name="toolTip">
        <string>Save and close</string>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="icon">
        <iconset resource="../brewken.qrc">
         <normaloff>:/images/filesave.svg</normaloff>:/images/filesave.svg</iconset>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="pushButton_cancel">
       <property name="toolTip">
        <string>Discard and close</string>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="icon">
        <iconset resource="../brewken.qrc">
         <normaloff>:/images/exit.svg</normaloff>:/images/exit.svg</iconset>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>BtGenericEdit</class>
   <extends>QLineEdit</extends>
   <header>BtLineEdit.h</header>
  </customwidget>
  <customwidget>
   <class>BtStringEdit</class>
   <extends>QLineEdit</extends>
   <header>BtLineEdit.h</header>
  </customwidget>
  <customwidget>
   <class>BtTimeEdit</class>
   <extends>QLineEdit</extends>
   <header>BtLineEdit.h</header>
   <slots>
    <slot>lineChanged(Measurement::UnitSystem const *, Measurement::UnitSystem::RelativeScale)</slot>
   </slots>
  </customwidget>
  <customwidget>
   <class>BtTimeLabel</class>
   <extends>QLabel</extends>
   <header>BtLabel.h</header>
   <slots>
    <signal>changedUnitSystemOrScale(Measurement::UnitSystem const *, Measurement::UnitSystem::RelativeScale)</signal>
   </slots>
  </customwidget>
  <customwidget>
   <class>BtMassEdit</class>
   <extends>QLineEdit</extends>
   <header>BtLineEdit.h</header>
   <slots>
    <signal>textModified()</signal>
    <slot>lineChanged()</slot>
    <slot>lineChanged(Measurement::UnitSystem const *, Measurement::UnitSystem::RelativeScale)</slot>
   </slots>
  </customwidget>
  <customwidget>
   <class>BtMassLabel</class>
   <extends>QLabel</extends>
   <header>BtLabel.h</header>
   <slots>
    <signal>changedUnitSystemOrScale(Measurement::UnitSystem const *, Measurement::UnitSystem::RelativeScale)</signal>
    <slot>popContextMenu(QPoint)</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>tabWidget_editor</tabstop>
  <tabstop>lineEdit_name</tabstop>
  <tabstop>lineEdit_alpha</tabstop>
  <tabstop>comboBox_use</tabstop>
  <tabstop>comboBox_type</tabstop>
  <tabstop>comboBox_form</tabstop>
  <tabstop>lineEdit_beta</tabstop>
  <tabstop>lineEdit_humulene</tabstop>
  <tabstop>lineEdit_caryophyllene</tabstop>
  <tabstop>lineEdit_cohumulone</tabstop>
  <tabstop>lineEdit_myrcene</tabstop>
  <tabstop>lineEdit_time</tabstop>
  <tabstop>lineEdit_HSI</tabstop>
  <tabstop>lineEdit_origin</tabstop>
  <tabstop>pushButton_new</tabstop>
  <tabstop>pushButton_save</tabstop>
  <tabstop>pushButton_cancel</tabstop>
  <tabstop>textEdit_substitutes</tabstop>
  <tabstop>textEdit_notes</tabstop>
 </tabstops>
 <resources>
  <include location="../brewken.qrc"/>
 </resources>
 <connections/>
</ui>
